src/features/vote/proposals/list/ProposalListItem.tsx

```diff
@@ -40,17 +47,66 @@ export default function ProposalListItem({ proposal }: Props) {
     return 'A';
   }, [authorName]);
 
+  const numberUpvotes = useMemo(() => {
+    // if status is not open, return total number of votes
+    if (!isOpen) {
+      return 72; // TODO: Replace with actual vote total
+    }
+    // if status is open, only show the user's votes
+    // and if vote is below zero return 0 for postive votes
+    const userVote = parseInt(voteValue, 10);
+
+    return userVote > 0 ? voteValue : 0;
+  }, [voteValue, isOpen]);
+  const numberDownvotes = useMemo(() => {
+    // if status is not open, return total number of votes
+    if (!isOpen) {
+      return 18; // TODO: Replace with actual vote total
+    }
+    // if status is open, only show the user's votes
+    // and if vote is above zero return 0 for negative votes
+    const userVote = parseInt(voteValue, 10);
+
+    return userVote < 0 ? voteValue : 0;
+  }, [voteValue, isOpen]);
+
+  const voteRequest = useFetchPost(
+    `https://vote.tulsawebdevs.org/proposals/${proposal.id}/vote`,
+  );
+
+  const handleVote = useProtectedFunction(
+    useDebounce((value: Vote) => {
+      const votePayload: VotePayload = {
+        initiativeId: proposal.id.toString(10),
+        vote: value,
+        comment: '',
+        authorId: proposal.authorId,
+        authorName: proposal.authorName,
+        authorEmail: proposal.authorEmail,
+      };
+
+      setVoteValue(value);
+      voteRequest(votePayload).catch(() => {
+        toast.error('Could not cast vote. Please try again.');
+      });
+    }),
+    {
+      unauthorizedMessage: 'You do not have permission to vote.',
+    },
+  );
```

Comment:
ISSUE: We probably want to take the `setVoteValue(value)` call out of the debounce to improve perceived performance. As it is, some clicks are 'ignored' from the user's perspective.
FIX: If I'm not mistaken, we should just be able to lift the function call out of the debounce hook without causing the value to fall out of sync with what's sent in the request.
NOTE(non-blocking): Normally we'd want to store the user's initial vote in memory and only update it when we get a success confirmation from the server. That way we can roll back the UI if the vote request fails. No need to worry about it here, though; i'm researching some libraries to use for data fetching that should abstract most of the query management cycle away.

Reply:
That makes sense. I thought it was awkward to see a slight delay. I've lifted up setting the vote value.